src/bridge/proxy_stream.ts

```diff
@@ -46,8 +53,37 @@ registerBridge("readProxyStream", function (ctx: Context, config: Config, ref: i
   stream.on("error", function (err: Error) {
     ctx.applyCallback(cb, ["error", err.toString()])
   })
-  stream.on("data", function (data: Buffer) {
+  /*stream.on("data", function (data: Buffer) {
+    console.log("data length:", data.length, stream.isPaused())
+    proxyable.buffered.push(data)
+    console.log("Buffer length: ", proxyable.buffered.length)
     ctx.applyCallback(cb, ["data", transferInto(data)])
-  })
-  setImmediate(() => stream.resume())
+  })*/
+  //setImmediate(() => stream.resume())
+})
+registerBridge("readProxyStream", function(ctx: Context, config: Config, ref: ivm.Reference<ProxyStream>, cb: ivm.Reference<Function>){
+  ctx.addCallback(cb)
+  const proxyable = ref.deref({ release: true })
+  const stream = proxyable.stream
+
+  let attempts = 0
+  const tryRead = function(){
+    attempts += 1
+    let chunk = stream.read(1024 * 1024)
+    let length = -1
+    if(chunk) length = chunk.byteLength
+
+    if(chunk){
+      proxyable.buffered.push(chunk)
```

Comment:
So we will buffer all chunks without limit? We probably want to put a limit here just in case.

Ideally we would want to stream both ways and that would remove the need to buffer anything.

Reply:
Ah yeah that's something we should limit.

I'm actually only buffering so we don't lose the first chunk. After that this buffer is irrelevant, I think I can mark it dirty and throw an error if we try and use the reference again after multiple chunk reads (which is when we assume that v8env is now the stream owner). I'll fix that in a bit.